### ISSUE TITLE
Adding generic MCOParameter subtypes

### ISSUE THREAD

**Issue by Corwinpro** [force-h2020/force-bdss#240]
Currently example plugins are responsible for creation of custom `MCOParameter` types, relevant to their need. At this stage, we can start thinking about adding new generic `MCOParameter`s types. Essentially, these are already implemented in ITWM example, and some other examples. We would benefit from reusing those, and it would be easier to maintain them as a part of the BDSS core.

I would suggest these types:

```
# constant-valued dummy parameter
class FixedMCOParameter(BaseMCOParameter): pass

# numerical parameter with discrete values
class ListedMCOParameter(BaseMCOParameter): pass

# numerical parameter with continuous values in some range
class RangedMCOParameter(BaseMCOParameter): pass

# categorical unordered parameters
class CategoricalMCOParameter(BaseMCOParameter): pass
```

We can also think about vector-values parameters, if needed.

**Comment by Corwinpro** [comment]
Note: there should be an obvious API to communicate the type of the `MCOParameter` to external data types, e.g. `nevergrad`.

**Comment by Corwinpro** [comment]
Proposed `MCOParameters` are implemented in #242
